### PR TITLE
Update Owe implementation

### LINKED ISSUE #276: Owe implementation lacks description

### PR DESCRIPTION
**Pull request by TY1Fan**
closes #276

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -211,11 +211,27 @@ Similarly, how an AddCommand operation goes through the `Model` component is sho
...
+If the constraints are not met, the `OweCommandParser` will throw a `ParseException` with an error message indicating the constraint that was violated.
+Otherwise, a new instance of `OweCommand` is then created with the values of `INDEX` and `HOURS_OWED` parsed by `OweCommandParser`.
+
+On execution, `OweCommand` first queries the supplied model for the student to be updated using the `INDEX`. 
+
+Then, `OweCommand` calculates the amount of tuition fee owed and checks if the total amount owed by the student exceeds the limit of `9999999.99`. If it exceeds, `OweCommand` will throw a `CommandException` with an error message indicating that limit was violated.
+
+Finally, `OweCommand` updates the total amount of tuition fee owed by the student by creating a new `Student` instance with updated fields to replace the outdated `Student` instance in the model.
+
+The owe command is part of UGTeach's payment tracking feature. The following activity diagram summarizes what happens when a user wants to track payment after a lesson:
```
Comment (Feng1231) [inline docs/DeveloperGuide.md:234]: I personally think that this statement can be at the top.
`The owe command is part of UGTeach's payment tracking feature.`

**Review by Feng1231 — COMMENTED** [review]
LGTM, except I think one statement can be moved to the top

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (TY1Fan) [reply docs/DeveloperGuide.md]: Thank you for ur suggestion. Ive shifted this line up

**Review by Feng1231 — APPROVED** [review]
LGTM

**Pull request merged**